### DISCUSSION TITLE
Add `get_one` to `Session`

### DISCUSSION THREAD

**Discussion by samtoffler** [sqlalchemy/sqlalchemy#10291] [Ideas]
I'm new to SQLAlchemy, so apologies if I've missed this somewhere in my research.

A common pattern I've seen is the following:

```
foo = db.session.get(Foo, foo_id)

if foo is None:
  raise Exception("Foo not found")
```

It would be nice if there was something like `Session.get_one()`, similar to [`Result.one()`](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Result.one), that raises a `NoResultFound` exception in the event that no instance is found. I could use `Session.query().one()`, but would like to keep the performance benefits of `get` as described in the [docs](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.get).

**Comment by CaselIT** [comment]
Hi,

I was thinking about something similar yesterday too, but I was wondering if having a flag in `get` would be a better option, something like `get(cls, the_id, required=True)`

But `get_one` is probably more in line with the rest of sqlalchemy api

**Comment by zzzeek** [comment]
ooooook.    We can do this one.   I am very conservative about too many "method_that_puts_things_together" because you get an explosion of methods which degrades composability, but `get()` is indeed special, so this one should be fine, if folks want to send PRs.

**Reply by CaselIT** [reply]
just to clarify you are suggesting `get_one` right?

**Reply by zzzeek** [reply]
oh sorry, yes get_one(), though  then we have to send all the arguments through....

**Reply by CaselIT** [reply]
does `scalars` list all args or uses `**kw`?

**Reply by CaselIT** [reply]
kinda a mix, lists some then has **kw. I guess we can do the same

**Comment by ziima** [comment]
`scalars().one()` raises `NoResultFound` if the result is not found, e.g.
```python
(await session.execute(select(MyModel).filter_by(id=148357843))).scalars().one()
# NoResultFound: No row was found when one was required
```

**Reply by CaselIT** [reply]
indeed. the objective of `get_one` would be to have the same behaviour but for `get`
